docs/reference/build_failures.md

```diff
@@ -1,62 +1,302 @@
-  too, especially for scientific code. Example: torch 1.12.0 support Python 3.7 to 3.10
-  (https://pypi.org/project/torch/1.12.0/#files), while numpy 2.1.0 supports Python 3.10 to 3.13
-  (https://numpy.org/doc/stable/release/2.1.0-notes.html#numpy-2-1-0-release-notes), so both
-  together mean you need Python 3.10 (or upgrade torch).
-- If locking fails due to building a package from a platform you do not support, consider
-  [declaring resolver environments](https://docs.astral.sh/uv/reference/settings/#environments) with
-  your supported platforms.
-- If you support a large range of Python versions, consider using markers to use older versions for
-  older Python versions and newer versions for newer Python version. In the example, numpy tends to
-  support four Python minor version at a time, so to support Python 3.8 to 3.13, the versions need
-  to be split:
-
-      ```
-      numpy>=1.23; python_version >= "3.10"
-      numpy<1.23; python_version < "3.10"
-      ```
-
-- If locking fails due to building a package from a different platform, as an escape hatch you can
-  [provide dependency metadata manually](https://docs.astral.sh/uv/reference/settings/#dependency-metadata).
-  As uv can not verify this information, it is important to specify correct metadata in this
-  override.
+https://pypi.org/project/numpy/2.1.1.md#files. Wheels with `...-py3-none-any.whl` filenames work
+everywhere, others have the operating system and platform in the filename. In the linked `numpy`
+example, you can see that there are pre-built distributions for Python 3.10 to 3.13 on MacOS, Linux
+and Windows.
+
+## Common build failures
+
+The following examples demonstrate common build failures and how to resolve them.
+
+### Command is not found
+
+If the build error mentions a missing command, for example, `gcc`:
+
+<!-- docker run --platform linux/x86_64 -it ghcr.io/astral-sh/uv:python3.10-bookworm-slim /bin/bash -c "uv pip install --system pysha3==1.0.2" -->
+
+```hl_lines="17"
+× Failed to download and build `pysha3==1.0.2`
+╰─▶ Build backend failed to build wheel through `build_wheel` (exit status: 1)
+
+    [stdout]
+    running bdist_wheel
+    running build
+    running build_py
+    creating build/lib.linux-x86_64-cpython-310
+    copying sha3.py -> build/lib.linux-x86_64-cpython-310
+    running build_ext
+    building '_pysha3' extension
+    creating build/temp.linux-x86_64-cpython-310/Modules/_sha3
+    gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -DPY_WITH_KECCAK=1 -I/root/.cache/uv/builds-v0/.tmpxAJdUa/include -I/usr/local/include/python3.10 -c Modules/_sha3/sha3module.c -o
+    build/temp.linux-x86_64-cpython-310/Modules/_sha3/sha3module.o
+
+    [stderr]
+    error: command 'gcc' failed: No such file or directory
+```
+
+Then, you'll need to install it with your system package manager, e.g., to resolve the error above:
+
+```console
+$ apt-get install gcc
```

Comment:
While it's a good example for "If X is missing, try `apt install X`", we should consider guiding ubuntu and debian user to `build-essential`, it's true to its name and has everything that you usually need for C/C++ packages